doc/contributing/gn-build.md

```diff
@@ -0,0 +1,112 @@
+# GN Build
+
+Similar to GYP, GN is a build system designed for building Chromium. The
+official builds of Node.js are built with GYP, but GN build files are also
+provided as an unofficial alternative build system.
+
```

Comment:
It would be better to emphasize a bit more about the support status:

```suggestion

The GN build files only support a subset of the Node.js build
configurations. It's not required for all pull requests to Node.js to
support GN builds, and Node.js's own CI does not currently test GN
builds, though Node.js welcomes pull requests that improve GN support
or fix breakages introduced by other pull requests.
```